packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -213,6 +213,7 @@ function hydrateStaticElement(elm: Node, vnode: VStatic, renderer: RendererAPI):
     }
 
     vnode.elm = elm;
+    applyEventListeners(vnode, renderer);
```

Comment:
I'm wondering if we need any new tests for hydration part of this? Maybe the existing tests would already cover it?

Reply:
Good point! The existing tests did actually catch this, but just to be safe, I added some more: ee964c9e1